src/status_im2/contexts/chat/messages/content/image/view.cljs

```diff
@@ -14,23 +15,28 @@
 
 (defn image-message
   [index {:keys [content image-width image-height message-id] :as message} context on-long-press]
-  (let [dimensions (calculate-dimensions (or image-width 1000) (or image-height 1000))
-        text       (:text content)]
-    (fn []
-      (let [shared-element-id (rf/sub [:shared-element-id])]
-        [rn/touchable-opacity
-         {:active-opacity 1
-          :key            message-id
-          :style          {:margin-top (when (> index 0) 10)}
-          :on-long-press  on-long-press
-          :on-press       (fn []
-                            (rf/dispatch [:chat.ui/update-shared-element-id message-id])
-                            (js/setTimeout #(rf/dispatch [:navigate-to :lightbox
-                                                          {:messages [message] :index 0}])
-                                           100))}
-         (when (and (not= text "placeholder") (= index 0))
-           [rn/view {:style {:margin-bottom 10}} [text/text-content message context]])
-         [fast-image/fast-image
-          {:source    {:uri (:image content)}
-           :style     (merge dimensions {:border-radius 12})
-           :native-ID (when (= shared-element-id message-id) :shared-element)}]]))))
+  [:f>
+   (fn []
+     (let [insets     (safe-area/use-safe-area)
+           dimensions (calculate-dimensions (or image-width 1000) (or image-height 1000))
+           text       (:text content)]
+       (fn []
+         (let [shared-element-id (rf/sub [:shared-element-id])]
+           [rn/touchable-opacity
+            {:active-opacity 1
+             :key            message-id
+             :style          {:margin-top (when (> index 0) 10)}
+             :on-long-press  on-long-press
+             :on-press       (fn []
+                               (rf/dispatch [:chat.ui/update-shared-element-id message-id])
+                               (js/setTimeout #(rf/dispatch [:navigate-to :lightbox
+                                                             {:messages [message]
+                                                              :index    0
+                                                              :insets   insets}])
+                                              100))}
```

Comment:
@ilmotta so this delay is needed to ensure that the `update-shared-element-id` event has dispatched before navigating, otherwise the shared element transition will not execute. Running it without the delay causes it sometimes work and sometimes not. Is there a way in `re-frame` to dispatch an event right after another event? (like a `.then` call)

Reply:
Thanks for your reply @ilmotta. I have opened an issue to investigate this issue further: https://github.com/status-im/status-mobile/issues/15358

Is there a specific function in re-frame to make an event "atomic" ?

Reply:
Great I will look into it. Thanks for sharing @ilmotta